src/test/java/com/google/sps/GetPlayerNameFromDatabaseTest.java

```diff
@@ -0,0 +1,118 @@
+  private static final String USER_ID_KEY_PATH =
+      "com.google.appengine.api.users.UserService.user_id_key";
+  private static Map<String, Object> USER_ID_CONFIG =
+      new HashMap<String, Object>() {
+        {
+          put(USER_ID_KEY_PATH, CURR_USER_ID);
+        }
+      };
+  // creating a test user service to mock the doget method
+  private final LocalUserServiceTestConfig localUserServiceTestConfig =
+      new LocalUserServiceTestConfig().setOAuthUserId(CURR_USER_ID);
+  private final LocalDatastoreServiceTestConfig localDatastoreServiceTestConfig =
+      new LocalDatastoreServiceTestConfig();
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(localUserServiceTestConfig, localDatastoreServiceTestConfig)
+          .setEnvIsAdmin(false)
+          .setEnvAuthDomain(AUTH_DOMAIN)
+          .setEnvEmail(EMAIL)
+          .setEnvIsLoggedIn(true)
+          .setEnvAttributes(USER_ID_CONFIG);
+  private DatastoreService localDatastore;
+  private UserService localUserService;
+  private PlayerDatabase playerDatabase;
+  private static final String LOGGED_OUT_EXCEPTION =
+      "Player is currently logged out. Cannot process null user.";
+  private static final String DISPLAY_NAME = "admin";
+  private static final String EMAIL = "admin@admin.com";
+  private static final Gson gson = new Gson();
+  private static GetPlayerNameFromDatabase getPayerNameServlet;
+
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+
+  @Rule public ExpectedException emptyName = ExpectedException.none();
+
+  @Before
+  public void setUp() throws Exception {
+    helper.setUp();
+    try {
+      this.localUserService = UserServiceFactory.getUserService();
+      this.localDatastore = DatastoreServiceFactory.getDatastoreService();
+      this.playerDatabase = new PlayerDatabase(this.localDatastore, this.localUserService);
+      this.getPayerNameServlet = new GetPlayerNameFromDatabase();
+      this.getPayerNameServlet.init();
+      MockitoAnnotations.initMocks(this);
+    } catch (Exception initialize) {
+      throw initialize;
+    }
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void doGet_GetPLayerNameFromDatabase() throws IOException, LoggedOutException {
+    User currentUser = this.localUserService.getCurrentUser();
+    Player player = createCurrentPlayer(this.playerDatabase, currentUser);
+    this.playerDatabase.addPlayerToDatabase(player);
+    this.playerDatabase.setEntityCurrentPageID(CURRENT_PAGE_ID);
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter printWriter = new PrintWriter(stringWriter);
+    when(this.response.getWriter()).thenReturn(printWriter);
+    this.getPayerNameServlet.doGet(this.request, this.response);
+    String result = stringWriter.toString();
+    String expected = gson.toJson(DISPLAY_NAME).toString();
+    Assert.assertEquals(expected, result);
+  }
+
```

Comment:
Add a test for logged out exception?

Reply:
Thank you for the feedback Implemented now